admin/javascript/LeftAndMain.Ping.js

```diff
@@ -21,16 +28,67 @@
 			 *
 			 * This function is called by prototype when it receives notification that the user was logged out.
 			 * It uses /Security/ping for this purpose, which should return '1' if a valid user session exists.
-			 * It redirects back to the login form if the URL is either unreachable, or returns '0'.
+			 * It opens the login form in a new window if the URL is either unreachable (doesn't return HTTP 200) or doesn't return the expected '1'.
 			 */
 			_setupPinging: function() {
+				var loginWindow;	// Reference to pop-up window generated if/when ping fails.
+				var dialog;			// Reference to message dialog, if one is ever opened.
+
 				var onSessionLost = function(xmlhttp, status) {
-					if(xmlhttp.status > 400 || xmlhttp.responseText == 0) {
-						// TODO will pile up additional alerts when left unattended
-						if(window.open('Security/login')) {
-							alert('Please log in and then try again');
-						} else {
-							alert('Please enable pop-ups for this site');
+					// See if we're getting a response that contains a request to re-authenticate. If so, do that now.
+					if (xmlhttp.getResponseHeader('X-Reauthenticate')) {
+						//TODO: Change to the following line when entwine bug is found/fixed.
+						// $('.cms-container').entwine('ss').showLoginDialog();
+						containerScope.showLoginDialog();
+
+					} else if(xmlhttp.responseText !== "1") {
+						// In this case, the server may not have even responded at all. To be safe, open a new window and
+						// show a default message requesting user to login, in case the login window is already open or
+						// opens successfully. This is necessary so we don't attempt to load a login iframe to a down
+						// server (better to do that in a new window instead).
+						var message = 'Please log in and then try again.';
+
+						if (!loginWindow || (loginWindow && loginWindow.closed)) {
+							// Window hasn't yet been opened (or has been closed but must be reopened again).
+							loginWindow = window.open('Security/login');
```

Comment:
We should be using the in-cms reauthentication here, not the default Security/login in another window.